cmd/nodecmd/whitelist.go

```diff
@@ -0,0 +1,170 @@
+		Use:   "whitelist <clusterName> [IP] ",
+		Short: "(ALPHA Warning) Whitelist IP for access to all nodes in a cluster",
+		Long: `(ALPHA Warning) This command is currently in experimental mode.
+
+The node whitelist command adds IP to cloud security access rules allowing it to access
+all nodes in the cluster via ssh or http.`,
+		SilenceUsage: true,
+		Args:         cobra.MinimumNArgs(1),
+		RunE:         whitelistIP,
+	}
+	return cmd
+}
+
+func whitelistIP(_ *cobra.Command, args []string) error {
+	var err error
+	clustersConfig := models.ClustersConfig{}
+	if app.ClustersConfigExists() {
+		clustersConfig, err = app.LoadClustersConfig()
+		if err != nil {
+			return err
+		}
+	}
+	if len(clustersConfig.Clusters) == 0 {
+		ux.Logger.PrintToUser("There are no clusters defined.")
+		return nil
+	}
+	clusterName := args[0]
+	if err := checkCluster(clusterName); err != nil {
+		return err
+	}
+
+	userIPAddress := ""
+	if len(args) > 1 {
+		userIPAddress = args[1]
+	} else {
+		userIPAddress, err = utils.GetUserIPAddress()
+		if err != nil {
+			ux.Logger.PrintToUser("Failed to get user IP address")
+		}
+		ux.Logger.PrintToUser("No IP provided. Detected your IP address as %s.", userIPAddress)
+	}
+	if !utils.IsValideIP(userIPAddress) {
+		return fmt.Errorf("invalid IP address: %s", userIPAddress)
+	}
+	type regionSecurityGroup struct {
+		cloud         string
+		region        string
+		securityGroup string
+	}
+	cloudSecurityGroupList := []regionSecurityGroup{}
+	clusterNodes, err := getClusterNodes(clusterName)
+	if err != nil {
+		return err
+	}
+	for _, node := range clusterNodes {
+		nodeConfig, err := app.LoadClusterNodeConfig(node)
+		if err != nil {
+			ux.Logger.PrintToUser("Failed to parse node %s due to %s", node, err.Error())
+		}
+		if slices.Contains(cloudSecurityGroupList, regionSecurityGroup{
+			cloud:         nodeConfig.CloudService,
+			region:        nodeConfig.Region,
+			securityGroup: nodeConfig.SecurityGroup,
+		}) {
+			continue
+		}
+		cloudSecurityGroupList = append(cloudSecurityGroupList, regionSecurityGroup{
+			cloud:         nodeConfig.CloudService,
+			region:        nodeConfig.Region,
+			securityGroup: nodeConfig.SecurityGroup,
+		})
+	}
+	if len(cloudSecurityGroupList) == 0 {
+		ux.Logger.PrintToUser("No nodes found in cluster %s", clusterName)
+		return nil
+	}
+
+	// GCP doesn't have regions  so we need to reduce it to only list of security groups
+	gcpSGList := []string{}
+	// whitelist IP
+	for _, cloudSecurityGroup := range cloudSecurityGroupList {
+		if cloudSecurityGroup.cloud == constants.GCPCloudService {
+			if !slices.Contains(gcpSGList, cloudSecurityGroup.securityGroup) {
+				gcpSGList = append(gcpSGList, cloudSecurityGroup.securityGroup)
+			}
+			continue
+		}
+		ux.Logger.PrintToUser("Whitelisting IP %s in %s cloud region %s", userIPAddress, cloudSecurityGroup.cloud, cloudSecurityGroup.region)
+		if cloudSecurityGroup.cloud == "" || cloudSecurityGroup.cloud == constants.AWSCloudService {
```

Comment:
why empty string is aws?

Reply:
this is similar to https://github.com/ava-labs/avalanche-cli/blob/8e3d1331a0ba0ad8b512946de10c277dc5b2bf30/cmd/nodecmd/dynamic_ips.go#L44. 
I assume this is for backwards compatibility when AWS CLoud was the only cloud supported